### ISSUE TITLE
Extend `new_type` to support "module type names"

### ISSUE THREAD

**Issue by Elv13** [lunarmodules/LDoc#334]
(this issue split of a discussion in a ticket regarding removing the sandbox)

Right now, when creating a new module type (like to extend `@classmod` or order your modules into categories), the name of the category is used in the URL. This is a problem when the category name has spaces because of the way most parser "find" urls/links in text. The links have space in them rather than `%20` when you copy them. This means markdown/forum/terminal links get cropped. In AwesomeWM, I work around this in the template by doing a search&replace of `_` by a space for rendering. Still, it's a hack and the bug should get fixed in ldoc.

**Comment by AntumDeluge** [comment]
This, and #337, are related to the discussion in #308, correct?

**Comment by alerque** [comment]
Yes @AntumDeluge. So are (to assorted degrees) #333, #335, and #336.
